modules/sql-engine/README.md

```diff
@@ -1,4 +1,67 @@
-#Apache Ignite SQL Engine Module
---------------------------
+# Apache Ignite SQL Engine
 
-Apache Ignite SQL Engine module provides Apache Calcite based query engine.
+## Threading Model
+
+Every Ignite node uses two thread pools for query processing.
+
+### Planning pool
+
+This pool is used to optimize queries and DML statements.
+
+The unit of work for this pool is a query that should be optimized. An optimization task is compute-heavy. Besides, the planning cycle is not optimized well enough, hence some tasks might require unreasonable amounts of time to complete, thus should be killed by a timeout.
+
+The size of this pool was chosen by a dice roll and hardcoded to 4, and should be moved to configuration in the future. The pool is named ‘sql-planning-pool’.
```

Comment:
I bet one will forget to fix this description when the configuration issue will be fixed.
Let's just mention the poll name and default size. WDYT?